### PR TITLE
Added the join table for visits

### PR DESCRIPTION
**Pull request by moise10r**
# In this PR, I did the following:
Create a table named vets with the following columns:
- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- age: integer
- date_of_graduation: date

Create a "join table" called specializations to handle this relationship.
Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit
- Insert the following data for vets:
- Vet William Tatcher is 45 years old and graduated on Apr 23rd, 2000.
- Vet Maisy Smith is 26 years old and graduated on Jan 17th, 2019.
- Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
- Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specialities:
- Vet William Tatcher is specialized in Pokemon.
- Vet Stephanie Mendez is specialized in Digimon and Pokemon.
- Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits:

Write queries to answer the following:
- Who was the last animal seen by William Tatcher?
- How many different animals did Stephanie Mendez see?
- List all vets and their specialties, including vets with no specialties.
- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
- What animal has the most visits to vets?
- Who was Maisy Smith's first visit?
- Details for most recent visit: animal information, vet information, and date of visit.
- How many visits were with a vet that did not specialize in that animal's species?
- What speciality should Maisy Smith consider getting? Look for the species she gets the most.

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -166,3 +166,91 @@ UPDATE animals SET owner_id = 3 WHERE name IN  ('Devimon' ,'Plantmon');
...
 COMMIT;
+
+
+BEGIN;
+INSERT INTO vets (name, age, date_of_graduation) 
+  VALUES ('William Tatcher',  45, '2000-04-23'),
+    ('Maisy Smith', 26, '2019-01-17'),
+    ('Stephanie Mendez', 64, '1981-05-04'),
+    ('Jack Harkness', 38, '2008-06-08');
+END;
```
Comment (Emmyn5600) [inline data.sql:177]: - [ ] Let us modify our insertion using this way, to make our queries more readable:

  `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('William Tatcher', 45, '2000-04-23');`

   `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Maisy Smith', 26, '2019-01-17');`

   `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Stephanie Mendez', 64, '2019-05-17');`

   `INSERT INTO vets (name, age, date_of_graduation)
   VALUES ('Jack Harknes', 38, '2008-06-08');`

File: data.sql [hunk 2]
```diff
@@ -166,3 +166,91 @@ UPDATE animals SET owner_id = 3 WHERE name IN  ('Devimon' ,'Plantmon');
...
+INSERT INTO specializations (vet_id, species_id) 
+  VALUES ((SELECT id FROM vets WHERE name='William Tatcher'),
+     (SELECT id FROM species WHERE name='Pokemon')),
+    ((SELECT id FROM vets WHERE name='Stephanie Mendez'),
+     (SELECT id FROM species WHERE name='Digimon')),
+    ((SELECT id FROM vets WHERE name='Stephanie Mendez'),
+     (SELECT id FROM species WHERE name='Pokemon')),
+    ((SELECT id FROM vets WHERE name='Jack Harkness'),
+     (SELECT id FROM species WHERE name='Digimon'));    
+END;
```
Comment (Emmyn5600) [inline data.sql]: - [ ] Let us use this way again to insert the data into specializations

INSERT INTO specializations (species_id, vet_id)

 VALUES ((SELECT id FROM vets WHERE name = 'William Tatcher'), (SELECT id FROM species WHERE name = 'Pokemon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Pokemon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Digimon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Jack Harkness'), (SELECT id FROM species WHERE name = 'Digimon'));

File: data.sql [hunk 3]
```diff
@@ -166,3 +166,91 @@ UPDATE animals SET owner_id = 3 WHERE name IN  ('Devimon' ,'Plantmon');
...
+    ((SELECT id FROM vets WHERE name='Maisy Smith'),
+    (SELECT id FROM animals WHERE name='Boarmon'),
+    '-2020-08-03'),
+    ((SELECT id FROM vets WHERE name='Stephanie Mendez'),
+    (SELECT id FROM animals WHERE name='Blossom'),
+    '2020-05-24'),
+    ((SELECT id FROM vets WHERE name='William Tatcher'),
+    (SELECT id FROM animals WHERE name='Blossom'),
+    '2021-01-11');
+END;
```
Comment (Emmyn5600) [inline data.sql]: - [ ] You can refer to the above way to insert the data into visits table, below is sample for one of the insertion data.

 `INSERT INTO visits (animal_id, vet_id, vist_date) VALUES ((SELECT id FROM animal WHERE name = 'Agumon'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2020-05-24');`

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @moise10r 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 

         - Please work on the review too

    Cheers and Happy coding!:clap::clap::clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @moise10r 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by moise10r** [comment]
Thank you for the feedback and for approving this PR

**Pull request merged**